query-node/mappings/src/content/nft.ts

```diff
@@ -1012,8 +1014,11 @@ export async function contentNft_NftBought({ event, store }: EventContext & Stor
   // specific event processing
 
   // load video
-  const video = await getRequiredExistingEntity(store, Video, videoId.toString(), 'Non-existing video was bought')
-
+  const video = await getRequiredExistingEntity(store, Video, videoId.toString(), 'Non-existing video was bought', [
+    'nft',
+    'nft.transactionalStatus',
+  ])
+  const { nft } = video
```

Comment:
Use `getNftFromVideo` instead of `getRequiredExistingEntity` wherever you need to get both `video` and `nft` objects.

It will allow you to simplify the line with `price: (nft?.transactionalStatus as TransactionalStatusBuyNow).price.toNumber(),` as well.